### DISCUSSION TITLE
Why isn't RemoteConfig finding my key?

### DISCUSSION THREAD

**Discussion by jefflovejapan** [firebase/firebase-ios-sdk#12905] [Q&A]
I'm using [Strategy 1](https://firebase.google.com/docs/remote-config/loading) (`await RemoteConfig.fetchAndActivate()` on load) in my iOS app, but the key whose value I set in the console is not showing up (`RemoteConfig.configValue(for:)` is giving an empty string, and `RemoteConfig.allKeys(from:)` is returning at best a small handful of the keys that are configured in the console.)

I understand that all of the guides recommend calling `RemoteConfig.setDefaults` to update the local state to reflect the current state of what's in the console at the time you deploy, I just don't understand why this should be necessary, and I worry that the values in `setDefaults` won't be overridden if I change them in the console.

Could someone explain whether this is the intended behavior for RemoteConfig? Is retrieving the value that's configured in the console contingent on `RemoteConfig.setDefaults` having already been called?

**Comment by quavodanceq** [comment]
Yes, it is intended behavior for Firebase Remote Config to use RemoteConfig.setDefaults to define the default values for parameters. This ensures that your app has a complete set of parameter values, even if the fetch from the server fails or if the values haven't been fetched yet.
When you call RemoteConfig.fetchAndActivate(), it fetches the parameter values from the server and activates them, overriding the local default values if they exist. If the fetch fails or if the key doesn’t exist in the remote config, the app will fall back to the local defaults set using setDefaults.
